### PR TITLE
snippets: Add snippet for the Zephyr Bluetooth Controller

### PR DESCRIPTION
**Pull request by rugeGerritsen**
By defining this snippet it becomes simpler to build an application for the Zephyr Bluetooth Controller in environments where this is not enabled in device tree by default.

It removes the need for adding boilerplate overlay files to applications.

### CODE REVIEW

File: snippets/bt-ll-sw-split/bt-ll-sw-split.overlay [hunk 1]
```diff
@@ -0,0 +1,13 @@
+&bt_hci_controller {
+	status = "okay";
+};
+
+&bt_hci_sdc {
+	status = "disabled";
+};
```
Comment (rugeGerritsen) [inline snippets/bt-ll-sw-split/bt-ll-sw-split.overlay:7]: Not sure what do with other device tree nodes. Maybe simply remove it from here? But then there are two nodes enabled which may be a bit confusing

Comment (aescolar) [reply snippets/bt-ll-sw-split/bt-ll-sw-split.overlay:7]: out of tree defined nodes can't be here, that for sure

Comment (rugeGerritsen) [reply snippets/bt-ll-sw-split/bt-ll-sw-split.overlay:7]: Discussed offline. Will remove that part and keep them both enabled. 

Follow up work would include removing controller selection. Need to think though how to handle that nicely when there are two nodes enabled

File: snippets/bt-ll-sw-split/README.rst [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+###############################################
+
+.. code-block:: console
+
+   west build -S bt-ll-sw-split [...]
+
+Overview
+********
+
+This snippet sets the default Bluetooth controller to the Zephyr Bluetooth LE Controller.
```
Comment (cvinayak) [inline snippets/bt-ll-sw-split/README.rst:13]:
```suggestion
This snippet selects the Zephyr Bluetooth LE Controller.
```

**Review by cvinayak — DISMISSED** [review]

File: snippets/bt-ll-sw-split/README.rst [hunk 3]
```diff
@@ -0,0 +1,21 @@
+.. _snippet-bt-ll-sw-split:
+
+Zephyr Bluetooth LE Controller (bt-ll-sw-split)
+###############################################
+
+.. code-block:: console
+
+   west build -S bt-ll-sw-split [...]
```
Comment (aescolar) [inline snippets/bt-ll-sw-split/README.rst:8]: Minor suggestion:
```suggestion
You can build with this snippet by following the instructions in :ref:`the snippets usage page<using-snippets>`. When building with `west`, you can do:

.. code-block:: console

   west build -S bt-ll-sw-split [...]
```

**Review by aescolar — DISMISSED** [review]
Thanks :)

**Review by nordicjm — COMMENTED** [review]
Bit confused by this, these are the default in zephyr, so why is a snippet needed that changes the defaults... to the defaults? If this is something needed for ncs, surely it can go downstream?

**Comment by rugeGerritsen** [comment]
> Bit confused by this, these are the default in zephyr, so why is a snippet needed that changes the defaults... to the defaults? If this is something needed for ncs, surely it can go downstream?

It can go downstream, but that will add additional maintenance. The idea behind moving it upstream is that we can explicitly enable it for cases where a sample or test requires the use of the Zephyr Bluetooth Controller.
(Should be documented in the commit message)

**Comment by cvinayak** [comment]
> Bit confused by this, these are the default in zephyr, so why is a snippet needed that changes the defaults... to the defaults? If this is something needed for ncs, surely it can go downstream?

Some of the bsim tests a Zephyr Bluetooth Controller specific and they either need to be blacklisted or have SNIPPET added due to the default controller in downstream DTS is not Zephyr Bluetooth Controller.

There need to be a follow up upstream PR to add snippets to Zephyr Bluetooth Controller specific tests, this way no blacklisting is needed downstream and these specific tests can continue to build (and be tested).

Corrections to my understanding are welcome.

**Review by aescolar — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Review by cvinayak — APPROVED** [review]

**Comment by aescolar** [comment]
Setting the trivial label due to the incoming rc1. The PR has been under review for 2 days in CET, and very soon it will have been for 2 days in PDT, but the time check seems to want to reach 24:00 GMT -8(?)

**Pull request merged**